helix-term/src/application.rs

```diff
@@ -1029,13 +1041,67 @@ impl Application {
                             .collect();
                         Ok(json!(result))
                     }
-                };
+                    MethodCall::ShowDocument(params) => {
+                        use helix_view::editor::Action;
+
+                        if params.external.unwrap_or(false) {
+                            match open::that(params.uri.as_str()) {
+                                Ok(_) => Ok(json!(lsp::ShowDocumentResult { success: true })),
+                                Err(err) => {
+                                    let err = format!(
+                                        "failed to open path externally: {:?}: {:?}",
+                                        params.uri, err
+                                    );
+                                    self.editor.set_error(err);
+                                    Ok(json!(lsp::ShowDocumentResult { success: false }))
+                                }
+                            }
+                        } else {
+                            match params.uri.to_file_path() {
+                                Err(_) => {
+                                    let err = format!(
+                                        "unable to convert URI to filepath: {}",
+                                        params.uri
+                                    );
+                                    self.editor.set_error(err);
+                                    Ok(json!(lsp::ShowDocumentResult { success: false }))
+                                }
+                                Ok(path) => {
+                                    match self.editor.open(&path, Action::Replace) {
+                                        Err(err) => {
+                                            let err = format!(
+                                                "failed to open path: {:?}: {:?}",
+                                                params.uri, err
+                                            );
+                                            self.editor.set_error(err);
+                                            Ok(json!(lsp::ShowDocumentResult { success: false }))
+                                        }
+                                        Ok(_) => {
+                                            if let Some(range) = params.selection {
+                                                let (view, doc) = current!(self.editor);
+                                                // TODO: convert inside server
+                                                if let Some(new_range) = lsp_range_to_range(
+                                                    doc.text(),
+                                                    range,
+                                                    language_server.offset_encoding(),
+                                                ) {
+                                                    doc.set_selection(
```

Comment:
I think this should receive a jumplist entry so that the old position on the document isn't lost if it was already open.

```suggestion
                                                    let jump = (doc.id(), doc.selection(view.id).clone());
                                                    view.jumps.push(jump);
                                                    doc.set_selection(
```